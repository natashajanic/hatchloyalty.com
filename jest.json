{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
