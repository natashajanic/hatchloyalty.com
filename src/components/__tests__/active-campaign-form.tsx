import { shallow, ShallowWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import ActiveCampaignForm, { IActiveCampaignFormProps } from '../active-campaign-form'
import { async } from 'q';

describe('ActiveCampaignForm', () => {
  let props: IActiveCampaignFormProps
  let mountedComponent: ShallowWrapper<any, any, ActiveCampaignForm>

  beforeEach(() => {
    props = {
      formID: '_form_1',
      formSrc: 'https://hatchloyalty.activehosted.com/f/embed.php?id=1'
    }
    mountedComponent = shallow(<ActiveCampaignForm {...props}/>)
  })

  describe('render', () => {
    it('renders a div with the provided form ID', async () => {
      expect(mountedComponent.find('div').length).toEqual(1)
      expect(mountedComponent.find('div').at(0).prop('className')).toEqual(props.formID)
    })

    it('renders a script tag with the provided form src', async () => {
      expect(mountedComponent.find('script').length).toEqual(1)
      expect(mountedComponent.find('script').at(0).prop('src')).toEqual(props.formSrc)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
