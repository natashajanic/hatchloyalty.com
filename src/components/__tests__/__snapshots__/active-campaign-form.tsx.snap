// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveCampaignForm render matches the expected snapshot 1`] = `
<div
  className="_form_1"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      charSet="utf-8"
      src="https://hatchloyalty.activehosted.com/f/embed.php?id=1"
      type="text/javascript"
    />
  </HelmetWrapper>
</div>
`;
